client/app/vps/upgrade/vps-upgrade.controller.js

```diff
@@ -0,0 +1,85 @@
+class VpsUpgradeCtrl {
+    constructor ($filter, $translate, $q, VpsService) {
+        this.$filter = $filter;
+        this.$translate = $translate;
+        this.$q = $q;
+        this.Vps = VpsService;
+
+        this.loaders = {
+            init: false
+        };
+
+        this.step = 0;
+        this.order = null;
+        this.selectedModel = {};
+        this.upgradesList = null;
+        this.agree = {
+            value: false
+        };
+    }
+
+    $onInit () {
+        this.loaders.init = true;
+        this.loadNextStep();
+    }
+
+    canLoadNextStep (condition, step) {
+        if (condition) {
+            this.loadNextStep(step);
+        }
+    }
+
+    loadNextStep (step) {
+        this.step = step || this.step + 1;
+    }
+
+    getCurrentModel () {
+        return _.find(this.upgradesList, upgrade => upgrade.isCurrentModel === true);
+    }
+
+    loadUpgradesList () {
+        this.loaders.step1 = true;
+        if (!this.upgradesList) {
+            return this.Vps.upgradesList().then(data => {
+                this.upgradesList = data.results;
+                return data;
+            }).catch(data => {
+                this.resetAction();
```

Comment:
resetAction does not exist, but we don't want it anyway. It should display an error using CloudMessage instead.